crash pressing the "Call in progress notification"

### Steps to reproduce

1. Start a call
2. Tap "Call in progress notification", Tap to return to the call
3. Crash

### Outcome

No Crash


### Your phone model

op9p

### Operating system version

crdroid 9.7 on android 13

### Application version and app store

0.7.3 from gplay

### Homeserver

1.119.0+bookworm1

### Will you send logs?

Yes

### Are you willing to provide a PR?

No

```
11-20 19:32:19.517 24105 24105 E AndroidRuntime: FATAL EXCEPTION: main
11-20 19:32:19.517 24105 24105 E AndroidRuntime: Process: io.element.android.x, PID: 24105
11-20 19:32:19.517 24105 24105 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{io.element.android.x/io.element.android.features.call.impl.ui.ElementCallActivity}: coil.network.HttpException: lateinit property presenter has not been initialized
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3654)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3791)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5794)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5685)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:71)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:138)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.ClientTransactionHandler.executeTransaction(ClientTransactionHandler.java:61)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.ActivityThread.handleRelaunchActivityLocally(ActivityThread.java:5753)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2317)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7930)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:854)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: Caused by: coil.network.HttpException: lateinit property presenter has not been initialized
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at io.element.android.features.roomlist.impl.RoomListNode$View$10.invoke(SourceFile:2574)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.AnchoredGroupPath.CompositionLocalProvider(SourceFile:77)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.material3.TextKt.ProvideTextStyle(SourceFile:77)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.material3.ButtonKt$Button$2$1.invoke(SourceFile:1341)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.AnchoredGroupPath.CompositionLocalProvider(SourceFile:29)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.material3.MaterialThemeKt.MaterialTheme(SourceFile:248)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at io.element.android.compound.theme.ElementThemeKt$ElementTheme$2.invoke(SourceFile:1857)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.AnchoredGroupPath.CompositionLocalProvider(SourceFile:29)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at io.element.android.compound.theme.ElementThemeKt.ElementTheme(SourceFile:603)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at kotlin.io.TextStreamsKt.ElementThemeApp(SourceFile:237)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at io.element.android.features.logout.impl.LogoutViewKt$LogoutView$2.invoke(SourceFile:1909)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.ui.platform.ComposeView.Content(SourceFile:70)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.navigation.compose.NavHostKt$NavHost$4$2.invoke(SourceFile:391)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.AnchoredGroupPath.CompositionLocalProvider(SourceFile:29)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.ui.platform.CompositionLocalsKt.ProvideCommonCompositionLocals(SourceFile:303)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.material3.TabRowKt$TabRowImpl$1.invoke(SourceFile:61)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.AnchoredGroupPath.CompositionLocalProvider(SourceFile:29)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt.ProvideAndroidCompositionLocals(SourceFile:570)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.ui.platform.WrappedComposition$setContent$1$1.invoke(SourceFile:37)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.AnchoredGroupPath.CompositionLocalProvider(SourceFile:77)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.ui.platform.WrappedComposition$setContent$1$1.invoke(SourceFile:250)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.AnchoredGroupPath.invokeComposable(SourceFile:15)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.ComposerImpl.doCompose(SourceFile:197)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.CompositionImpl.composeContent(SourceFile:33)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.Recomposer.composeInitial$runtime_release(SourceFile:45)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.runtime.CompositionImpl.composeInitial(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.datastore.core.SimpleActor$1.invoke(SourceFile:461)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.ui.platform.AndroidComposeView.setOnViewTreeOwnersAvailable(SourceFile:7)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.ui.platform.WrappedComposition.setContent(SourceFile:12)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.compose.ui.platform.WrappedComposition.onStateChanged(SourceFile:19)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(SourceFile:26)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry.sync(SourceFile:330)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry.moveToState(SourceFile:77)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(SourceFile:15)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.lifecycle.ReportFragment$Companion.dispatch$lifecycle_runtime_release(SourceFile:27)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at androidx.lifecycle.ReportFragment$LifecycleCallbacks.onActivityPostCreated(SourceFile:10)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.Activity.dispatchActivityPostCreated(Activity.java:1400)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8370)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:8339)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1423)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3635)
11-20 19:32:19.517 24105 24105 E AndroidRuntime: 	... 17 more
```

--- a/features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/ElementCallActivity.kt
+++ b/features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/ElementCallActivity.kt
@@ -44,6 +44,7 @@ import io.element.android.features.call.impl.pip.PictureInPictureState
 import io.element.android.features.call.impl.pip.PipView
 import io.element.android.features.call.impl.services.CallForegroundService
 import io.element.android.features.call.impl.utils.CallIntentDataParser
+import io.element.android.libraries.architecture.Presenter
 import io.element.android.libraries.architecture.bindings
 import io.element.android.libraries.core.log.logger.LoggerTag
 import io.element.android.libraries.designsystem.theme.ElementThemeApp
@@ -62,7 +63,7 @@ class ElementCallActivity :
     @Inject lateinit var appPreferencesStore: AppPreferencesStore
     @Inject lateinit var pictureInPicturePresenter: PictureInPicturePresenter
 
-    private lateinit var presenter: CallScreenPresenter
+    private lateinit var presenter: Presenter<CallScreenState>
 
     private lateinit var audioManager: AudioManager
 
@@ -92,6 +93,10 @@ class ElementCallActivity :
         )
 
         setCallType(intent)
+        // If presenter is not created at this point, it means we have no call to display, the Activity is finishing, so return early
+        if (!::presenter.isInitialized) {
+            return
+        }
 
         if (savedInstanceState == null) {
             updateUiMode(resources.configuration)

--- a/features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/ElementCallActivity.kt
+++ b/features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/ElementCallActivity.kt
@@ -200,12 +200,14 @@ class ElementCallActivity :
             Timber.tag(loggerTag.value).d("Set the call type and create the presenter")
             webViewTarget.value = callType
             presenter = presenterFactory.create(callType!!, this)
+        } else if (callType == null) {
+            Timber.tag(loggerTag.value).d("Coming back from notification, do nothing")
         } else if (callType != currentCallType) {
             Timber.tag(loggerTag.value).d("User starts another call, restart the Activity")
             setIntent(intent)
             recreate()
         } else {
-            Timber.tag(loggerTag.value).d("Coming back from notification, do nothing")
+            Timber.tag(loggerTag.value).d("Other case, do nothing")
         }
     }
 

--- a/features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/ElementCallActivity.kt
+++ b/features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/ElementCallActivity.kt
@@ -198,21 +198,26 @@ class ElementCallActivity :
                 ?: intent.dataString?.let(::parseUrl)?.let(::ExternalUrl)
         }
         val currentCallType = webViewTarget.value
-        if (currentCallType == null && callType == null) {
-            Timber.tag(loggerTag.value).d("Re-opened the activity but we have no url to load or a cached one, finish the activity")
-            finish()
-        } else if (currentCallType == null) {
-            Timber.tag(loggerTag.value).d("Set the call type and create the presenter")
-            webViewTarget.value = callType
-            presenter = presenterFactory.create(callType!!, this)
-        } else if (callType == null) {
-            Timber.tag(loggerTag.value).d("Coming back from notification, do nothing")
-        } else if (callType != currentCallType) {
-            Timber.tag(loggerTag.value).d("User starts another call, restart the Activity")
-            setIntent(intent)
-            recreate()
+        if (currentCallType == null) {
+            if (callType == null) {
+                Timber.tag(loggerTag.value).d("Re-opened the activity but we have no url to load or a cached one, finish the activity")
+                finish()
+            } else {
+                Timber.tag(loggerTag.value).d("Set the call type and create the presenter")
+                webViewTarget.value = callType
+                presenter = presenterFactory.create(callType, this)
+            }
         } else {
-            Timber.tag(loggerTag.value).d("Other case, do nothing")
+            if (callType == null) {
+                Timber.tag(loggerTag.value).d("Coming back from notification, do nothing")
+            } else if (callType != currentCallType) {
+                Timber.tag(loggerTag.value).d("User starts another call, restart the Activity")
+                setIntent(intent)
+                recreate()
+            } else {
+                // Starting the same call again, should not happen, the UI is preventing this. But maybe when using external links.
+                Timber.tag(loggerTag.value).d("Starting the same call again, do nothing")
+            }
         }
     }
 

